sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsynReadWithMultipleClients.java

```diff
@@ -0,0 +1,295 @@
+                        .connectionPolicy(configuration.getConnectionPolicy())
+                        .consistencyLevel(configuration.getConsistencyLevel())
+                        .connectionReuseAcrossClientsEnabled(true)
+                        .buildAsyncClient();
+                    List<PojoizedJson> docsToRead = new ArrayList<>();
+                    if (!configuration.isDeleteCollections()) {
+                        CosmosAsyncDatabase cosmosAsyncDatabase = asyncClient.createDatabaseIfNotExists(this.configuration.getDatabaseId()).block().getDatabase();
+                        CosmosAsyncContainer cosmosAsyncContainer = cosmosAsyncDatabase.createContainerIfNotExists(configuration.getCollectionId(), PARTITION_KEY, 100000).block().getContainer();
+                        String partitionKey = cosmosAsyncContainer.read().block().getProperties().getPartitionKeyDefinition()
+                            .getPaths().iterator().next().split("/")[1];
+                        String dataFieldValue = RandomStringUtils.randomAlphabetic(this.configuration.getDocumentDataFieldSize());
+                        ArrayList<Flux<PojoizedJson>> createDocumentObservables = new ArrayList<>();
+
+                        for (int i = 0; i < this.configuration.getNumberOfPreCreatedDocuments(); i++) {
+                            String uuid = UUID.randomUUID().toString();
+                            com.azure.cosmos.benchmark.PojoizedJson newDoc = generateDocument(uuid, dataFieldValue, partitionKey);
+
+                            Flux<PojoizedJson> obs = cosmosAsyncContainer.createItem(newDoc).map(resp -> {
+                                    com.azure.cosmos.benchmark.PojoizedJson x =
+                                        resp.getItem();
+                                    return x;
+                                }
+                            ).flux();
+                            createDocumentObservables.add(obs);
+                        }
+                        docsToRead = Flux.merge(Flux.fromIterable(createDocumentObservables), 100).collectList().block();
+
+                        logger.info("Client have been initialized with data created for host {}", hostAndKey[0]);
+                    } else {
+                        logger.info("Client have been initialized with host {}", hostAndKey[0]);
+                    }
+                    clientDocsMap.put(asyncClient, docsToRead);
+                }
+            }
+            MeterRegistry registry = configuration.getAzureMonitorMeterRegistry();
+
+            if (registry != null) {
+                BridgeInternal.monitorTelemetry(registry);
+            }
+
+            registry = configuration.getGraphiteMeterRegistry();
+
+            if (registry != null) {
+                BridgeInternal.monitorTelemetry(registry);
+            }
+
+            logger.info("Total number of client created for ReadThroughputWithMultipleClient {}",clientDocsMap.size());
```

Comment:
codestyle: space after comma
please use intellij autoformatting on this new file.